### PR TITLE
moving to generic tag node

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: packages/better-php-doc-parser/src/PhpDocNode/AbstractTagValueNode.php [hunk 1]
```diff
@@ -293,4 +303,20 @@ private function createQuotedKeyPattern(?string $silentKey, string $key, string
...
+     * @param object|mixed[] $annotationOrItems
+     */
+    private function resolveItems($annotationOrItems): array
+    {
+        if (is_array($annotationOrItems)) {
+            return $annotationOrItems;
+        }
+
+        if (is_object($annotationOrItems)) {
+            return get_object_vars($annotationOrItems);
```
Comment (stephanvierkant) [inline packages/better-php-doc-parser/src/PhpDocNode/AbstractTagValueNode.php:317]: @TomasVotruba For some reason, this breaks Rector in my application. A `Symfony\Component\Routing\Annotation\Route` object returns `[]` in my case.

Comment (stephanvierkant) [reply packages/better-php-doc-parser/src/PhpDocNode/AbstractTagValueNode.php:317]: I'll try to find out why it fails in my application while the tests are OK, but it's hard to keep up with your development speed at this moment.

Comment (TomasVotruba) [reply packages/better-php-doc-parser/src/PhpDocNode/AbstractTagValueNode.php:317]: Thanks for such a fast testing, wow!

That's caused by Route annotation having private properties (made fake `public` in tests here for now) and `get_object_vars()` works only on public ones.

Do you have idea how to get into private properties of an object? Effectively.

Comment (TomasVotruba) [reply packages/better-php-doc-parser/src/PhpDocNode/AbstractTagValueNode.php:317]: This PR tests this bug, see: https://github.com/rectorphp/rector/pull/3285

**Pull request merged**